lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -381,6 +385,21 @@ data ByronWalletPostData mw = ByronWalletPostData
     , passphrase :: !(ApiT (Passphrase "encryption"))
     } deriving (Eq, Generic, Show)
 
+-- | A root private key, encrypted using a given passphrase.
+-- The underlying key should contain: private key, chain code, and public key
+newtype EncryptedRootXPrv =
+    EncryptedRootXPrv { getEncryptedRootXPrv :: ByteString }
+    deriving (Generic, Eq, Show)
+
+encryptedRootXPrvBytesLength :: Int
+encryptedRootXPrvBytesLength = 128
```

Comment:
If the value must be exactly 128 bytes long, then perhaps don't export the EncryptedRootXPrv constructor, instead provide one with `Maybe` error handling.

Reply:
As per previous suggestion, this could go and be pushed back onto the `xprv` constructor from `Cardano.Crypto` for constructing an XPrv from raw bytes.

Reply:
go for `ApiT XPrv` in https://github.com/input-output-hk/cardano-wallet/pull/1439/commits/07c2a3a7f869dccdee5af06cb2cc97ebca5814d0